### PR TITLE
Restart default txn on v2 checkpoints after 10 minutes of inactivity

### PR DESCRIPTION
**Pull request by nick4598**
This ends a read transaction marking blocks as unused allowing the daemon to better manage its cachefile. 

A 10 minute timer starts when db is opened as a lazy v2 checkpoint, this timer refreshes anytime refreshContainer is called which is called on every RPC Operation. If no refreshContainer call, then the timer will trigger ending the read transaction and marking blocks as unused.

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

File: core/backend/src/IModelDb.ts [hunk 1]
```diff
@@ -2752,21 +2758,30 @@ export class SnapshotDb extends IModelDb {
       throw err;
     }
 
+    // unref timer, so it doesn't prevent a process from shutting down.
+    snapshot._restartDefaultTxnTimer = setTimeout(() => {
+      snapshot.restartDefaultTxn();
```
Comment (kabentley) [inline core/backend/src/IModelDb.ts:2763]: this could benefit from a comment explaining why this is necessary.

Comment (nick4598) [reply core/backend/src/IModelDb.ts:2767]: Thanks, I added a comment to the timer defined on snapshotdb explaining it. Let me know your thoughts.

File: presentation/backend/src/test/PresentationRpcImpl.test.ts [hunk 2]
```diff
@@ -75,6 +82,8 @@ describe("PresentationRpcImpl", () => {
 
     const imodelTokenMock = moq.Mock.ofType<IModelRpcProps>();
     const imodelMock = moq.Mock.ofType<IModelDb>();
+    // need to define .then because mocks aren't 'thenable'. async functions which return IModelDb hang without this.
+    imodelMock.setup((x: any) => x.then).returns(() => undefined);
```
Comment (saskliutas) [inline presentation/backend/src/test/PresentationRpcImpl.test.ts:86]: You could use `configureForPromiseResult` helper function for this. https://github.com/iTwin/itwinjs-core/blob/8ef85dabc4b5665c69aa1b3abeacc594c66cb314/presentation/common/src/test/_helpers/Mocks.ts#L16

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: presentation/backend/src/test/PresentationRpcImpl.test.ts (hunk 2 above)
Comment (nick4598) [reply presentation/backend/src/test/PresentationRpcImpl.test.ts]: Thanks, changed to use that.

File: presentation/backend/src/presentation-backend/PresentationRpcImpl.ts [hunk 3]
```diff
@@ -106,10 +106,12 @@ export class PresentationRpcImpl extends PresentationRpcInterface implements IDi
     return Presentation.getManager(clientId);
   }
 
-  private getIModel(token: IModelRpcProps): IModelDb {
+  private async getIModel(accessToken: AccessToken, token: IModelRpcProps): Promise<IModelDb> {
     let imodel: IModelDb;
     try {
       imodel = IModelDb.findByKey(token.key);
+      // call refreshContainer, just in case this is a V2 checkpoint whose sasToken is about to expire, or its default transaction is about to be restarted.
+      await imodel.refreshContainer(accessToken);
```
[the remark this answers is not in the document]
Comment (nick4598) [reply presentation/backend/src/presentation-backend/PresentationRpcImpl.ts]: Removed it

**Review by saskliutas — APPROVED** [review]

**Comment by nick4598** [comment]
@mergifyio backport release/4.1.x

**Pull request merged**